blocksuite/framework/std/src/clipboard/clipboard.ts

```diff
@@ -267,45 +267,44 @@ export class Clipboard extends LifeCycleWatcher {
       items: Record<string, unknown>
     ) => Promise<Record<string, unknown>> | Record<string, unknown>
   ) {
-    const _items = {
-      'text/plain': '',
-      'text/html': '',
-      'image/png': '',
-    };
+    const clipboardItems: Record<string, Blob> = {};
 
-    const items = await updateItems(_items);
+    const items = await updateItems({});
 
     const text = items['text/plain'] as string;
     const innerHTML = items['text/html'] as string;
-    const png = items['image/png'] as string | Blob;
+    const image = Object.entries(items).find(([key]) =>
+      key.startsWith('image/')
+    );
 
     delete items['text/plain'];
     delete items['text/html'];
-    delete items['image/png'];
 
-    const snapshot = lz.compressToEncodedURIComponent(JSON.stringify(items));
-    const html = `<div data-blocksuite-snapshot='${snapshot}'>${innerHTML}</div>`;
-    const htmlBlob = new Blob([html], {
-      type: 'text/html',
-    });
-    const clipboardItems: Record<string, Blob> = {
-      'text/html': htmlBlob,
-    };
-    if (text.length > 0) {
-      const textBlob = new Blob([text], {
-        type: 'text/plain',
-      });
-      clipboardItems['text/plain'] = textBlob;
+    if (image) {
+      const [type, blob] = image;
+
+      delete items[type];
+
+      if (typeof blob === 'string') {
+        clipboardItems[type] = new Blob([blob], { type });
+      } else if (blob instanceof Blob) {
+        clipboardItems[type] = blob;
+      }
     }
 
-    if (png instanceof Blob) {
-      clipboardItems['image/png'] = png;
-    } else if (png.length > 0) {
-      const pngBlob = new Blob([png], {
-        type: 'image/png',
-      });
-      clipboardItems['image/png'] = pngBlob;
+    if (text?.length > 0) {
+      const type = 'text/plain';
+      clipboardItems[type] = new Blob([text], { type });
     }
+
+    // If there are no items, fall back to snapshot.
+    if (innerHTML?.length > 0 || Object.keys(clipboardItems).length === 0) {
+      const type = 'text/html';
+      const snapshot = lz.compressToEncodedURIComponent(JSON.stringify(items));
+      const html = `<div data-blocksuite-snapshot='${snapshot}'>${innerHTML}</div>`;
```

Comment:
When `innerHTML === undefined` and `clipboardItems` is empty, you will get a string including a substring "undefeined", which may be rendered to users.